src/main/java/duke/Gui/Gui.java

```diff
@@ -0,0 +1,131 @@
+package duke.Gui;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Gui {
+    private static final String INDENT = "  ";
+    Scanner scan;
+
+    public Gui() {
+        this.scan = new Scanner(System.in);
+    }
+
+    public String showWelcome() {
+        return ("Hello! I'm duke.Duke\n" + "What can I do for you?");
+    }
+
+    public ArrayList<String> showBye() {
+        ArrayList<String> responseList = new ArrayList<>();
+        responseList.add("Bye!!! Hope to see you again real soon.");
+
+        return responseList;
+    }
+
+    public ArrayList<String> showDone() {
+        ArrayList<String> responseList = new ArrayList<>();
+        responseList.add("The following task has been marked done: ");
+
+        return responseList;
+    }
+
+    public String showError(String e) {
+        return (e);
+    }
+
+    /**
+     * prints out all the tasks.
+     * @param list list of tasks.
+     */
+    public ArrayList<String> listAllTasks(ArrayList<Task> list) {
+        ArrayList<String> responseList = new ArrayList<>();
+        responseList.add("Here are the tasks in your list: ");
+
+        int LENGTH_OF_LIST = list.size();
+        if (LENGTH_OF_LIST > 0) {
+            int counter = 1;
+            for (Task task : list) {
+                responseList.add("  " + counter + "." + task);
+                counter++;
+            }
+        } else {
+            responseList.remove(0);
+            responseList.add("No tasks found, add a task now!");
+        }
+        return responseList;
+    }
+
+    /**
+     * message after adding a task to list.
+     * @param task task to be added.
+     * @param tasksLeft integer value of tasks remaining in the list of tasks.
+     */
+    public ArrayList<String> addMessage(Task task, int tasksLeft) {
+        ArrayList<String> reponseList = new ArrayList<>();
+
+        reponseList.add("Got it, the following task has been added:\n" + INDENT + INDENT + task +
+                "\n" + INDENT + this.taskLeftMessage(tasksLeft));
+
+        return reponseList;
+    }
+
+    /**
+     * message after searching for term with Find
```

Comment:
Perhaps the description of the method could have been clearer? Can also summarise what the code does in the first sentence instead of saying what the code returns. Same for `deleteMessage` and `taskLeftMessage`.